src/components/MarkerPopup/MarkerPopup.jsx

```diff
@@ -121,13 +125,90 @@ export const MarkerContent = ({ place }) => {
     );
 };
 
-export const MarkerPopup = ({ place }) => (
-    <Marker position={place.position} key={place.metadata.UUID}>
-        <Popup>
-            <MarkerContent place={place} />
-        </Popup>
-    </Marker>
-);
+// TODO Rename MarkerPopup because it is not a popup for mobile
+
+// Maybe something like MarkerForDeviceType
+export const MarkerPopup = ({ place }) => {
+    const [open, setOpen] = useState(false);
+    const map = useMap();
+
+    const handleClickOpen = () => {
+        if (isMobile) {
+            const offset = 0.003;
+            const newLat = place.position[0] - offset;
+            map.panTo([newLat, place.position[1]], { duration: 0.5 });
+            console.log('clicked');
+            setOpen(true);
+        }
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    let contentForDeviceType;
+    let eventHandlersForDeviceType;
+
+    if (isMobile) {
+        eventHandlersForDeviceType = { click: handleClickOpen };
+
+        contentForDeviceType = (
+            <Dialog
+                open={open}
+                onClose={handleClose}
+                fullWidth
+                maxWidth="md"
+                style={{
+                    position: 'fixed',
+                    bottom: 0,
+                    margin: 0,
+                }}
+                PaperProps={{
+                    style: {
+                        position: 'fixed',
+                        bottom: 0,
+                        margin: 0,
+                        width: '100%',
+                        maxHeight: '50%',
+                    },
+                }}
+            >
+                <DialogTitle>
+                    <IconButton
+                        aria-label="close"
+                        onClick={handleClose}
+                        style={{ position: 'absolute', right: 8, top: 8 }}
+                    >
+                        <CloseIcon />
+                    </IconButton>
+                </DialogTitle>
+                <DialogContent>
+                    <MarkerContent place={place} />
+                </DialogContent>
+            </Dialog>
```

Comment:
this could be `MyExtraMobileDialog` and there could be side effect code along.

Reply:
and then you'd have simple:
`if is mobile return myextramobiledialog else return notsoextrapopup`

Reply:
could be new component (file) to clean it up. :)